### DISCUSSION TITLE
Send torque.

### DISCUSSION THREAD

**Discussion by Carmol1** [bitcraze/discussions#268] [Q&A]
Hi.

I want to make use of a offboard control law, for this reason, I would like to know if do you have a function how def send_setpoint(self, roll, pitch, yawrate, thrust), but to send the torque?.

I would appreciate your help.

Regards.

**Comment by tobbeanton** [comment]
Could you perhaps clarify a bit with what you mean by torque? There is a discussion about thrust in grams [here](https://github.com/orgs/bitcraze/discussions/114).

**Reply by Carmol1** [reply]
Hi.

Thank you for your help, in my control law I get the thrust and the torque in x, y and z as it is shown in the image
![SmartSelect_20220916-130310_Drive](https://user-images.githubusercontent.com/104467412/190724012-b8e6317d-69f5-4028-aac0-062527926881.jpg)
for this reason I asked the above.

Regards.

**Reply by tobbeanton** [reply]
I think it would be very unfeasible to control a motor that drives a propeller with torque. That is very applicable for e.g. a ground driving vehicle but not as much for a drone. Anyway the simple motor drivers on the Crazyflie does not measure current and rotational speed so they can not be controlled with torque. Possibly some brushless ESC has those capabilities.

**Reply by Carmol1** [reply]
Hi.

Thank you for your replying, to conclude with this issue, I would like to know if is it possible to send angular velocities to the crazyflie using the following equation?
![SmartSelect_20220912-224022_Chrome](https://user-images.githubusercontent.com/104467412/191164877-18cbc2a4-d0e6-4dd9-b21e-7b25354e262e.jpg)
where $u_{1}=T$, $u_2=\tau_{x}$, $u_3=\tau_{y}$ and $u_4=\tau_{z}$.

Regards.

**Reply by knmcguire** [reply]
Something like that has not been implemented yet in our python library, although the support from the firmware exists. Please check out this open ticket: https://github.com/bitcraze/crazyflie-lib-python/issues/152

**Reply by Carmol1** [reply]
Hi.

I will check the ticket, thank you for your help.

Regards.

**Reply by krichardsson** [reply]
Hi! If you think this answered your question, please consider clicking the "Mark as answer" button. Thanks!
